File: algorithm/merge-two-binary-trees/merge-two-binary-trees.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* mergeTrees(TreeNode* t1, TreeNode* t2) {
        if (t1 == nullptr && t2 == nullptr) {
            return nullptr;
        } else if (t1 == nullptr) {
            TreeNode* t = new TreeNode(t2->val);
            t->left = t2->left;
            t->right = t2->right;
            return t;
        } else if (t2 == nullptr) {
            TreeNode* t = new TreeNode(t1->val);
            t->left = t1->left;
            t->right = t1->right;
            return t;
        }
        TreeNode *t = new TreeNode(t1->val + t2->val);
        t->left = this->mergeTrees(t1->left, t2->left);
        t->right = this->mergeTrees(t1->right, t2->right);
        return t;
    }
};

File: algorithm/merge-two-binary-trees/merge-two-binary-trees.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func mergeTrees(t1 *TreeNode, t2 *TreeNode) *TreeNode {
	if t1 == nil && t2 == nil {
		return nil
	} else if t1 == nil {
		return &TreeNode{t2.Val, t2.Left, t2.Right}
	} else if t2 == nil {
		return &TreeNode{t1.Val, t1.Left, t1.Right}
	}
	return &TreeNode{
		Val:   t1.Val + t2.Val,
		Left:  mergeTrees(t1.Left, t2.Left),
		Right: mergeTrees(t1.Right, t2.Right),
	}
}
